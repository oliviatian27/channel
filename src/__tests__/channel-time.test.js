import React from 'react';
import { shallow,mount } from 'enzyme';
import moment from 'moment'

import ChannelTime from '../components/channel-list/channel-time';


describe('ChannelTime tests', () => {

  it('renders Channel-time', () => {
    const time = "2019-01-05 19:00:00";
    const wrapper = shallow(<ChannelTime time={time} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-time')).toBeDefined();
    expect(wrapper.find('.channel-time')).toMatchSnapshot();
  });

  it('displays correct start and end time based on props.time', () => {
    const time = "2018-01-20 03:00:00";
    const wrapper = mount(<ChannelTime time={time} />);

    // Expect the time text to be props.time
    expect(wrapper.find('time').at(0).text()).toBe(moment(time).format("LT [-]"));
    expect(wrapper.find('time').at(1).text()).toBe(moment(time).add(1,'hours').format(" LT [EDT]"));

    //update the time props
    const newTime = "2018-01-21 18:00:00";
    wrapper.setProps({time:newTime})

    expect(wrapper.find('time').at(0).text()).toBe(moment(newTime).format("LT [-]"));
    expect(wrapper.find('time').at(1).text()).toBe(moment(newTime).add(1,'hours').format(" LT [EDT]"));
    wrapper.unmount()
  });
})
