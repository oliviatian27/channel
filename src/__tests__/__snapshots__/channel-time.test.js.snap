// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelTime tests renders Channel-time 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelTime
      time="2019-01-05 19:00:00"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="LT [-]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
          <t
            add={
              Object {
                "hours": 1,
              }
            }
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format=" LT [EDT]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
        ],
        "className": "channel-time",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": "LT [-]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "add": Object {
              "hours": 1,
            },
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": " LT [EDT]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <t
              ago={false}
              calendar={false}
              decimal={false}
              element="time"
              filter={[Function]}
              format="LT [-]"
              fromNow={false}
              interval={60000}
              onChange={[Function]}
              titleFormat=""
              toNow={false}
              unit={null}
              unix={false}
              utc={false}
              withTitle={false}
            >
              2019-01-05 19:00:00
            </t>,
            <t
              add={
                Object {
                  "hours": 1,
                }
              }
              ago={false}
              calendar={false}
              decimal={false}
              element="time"
              filter={[Function]}
              format=" LT [EDT]"
              fromNow={false}
              interval={60000}
              onChange={[Function]}
              titleFormat=""
              toNow={false}
              unit={null}
              unix={false}
              utc={false}
              withTitle={false}
            >
              2019-01-05 19:00:00
            </t>,
          ],
          "className": "channel-time",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ago": false,
              "calendar": false,
              "children": "2019-01-05 19:00:00",
              "decimal": false,
              "element": "time",
              "filter": [Function],
              "format": "LT [-]",
              "fromNow": false,
              "interval": 60000,
              "onChange": [Function],
              "titleFormat": "",
              "toNow": false,
              "unit": null,
              "unix": false,
              "utc": false,
              "withTitle": false,
            },
            "ref": null,
            "rendered": "2019-01-05 19:00:00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "add": Object {
                "hours": 1,
              },
              "ago": false,
              "calendar": false,
              "children": "2019-01-05 19:00:00",
              "decimal": false,
              "element": "time",
              "filter": [Function],
              "format": " LT [EDT]",
              "fromNow": false,
              "interval": 60000,
              "onChange": [Function],
              "titleFormat": "",
              "toNow": false,
              "unit": null,
              "unix": false,
              "utc": false,
              "withTitle": false,
            },
            "ref": null,
            "rendered": "2019-01-05 19:00:00",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="LT [-]"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-01-05 19:00:00
        </t>,
        <t
          add={
            Object {
              "hours": 1,
            }
          }
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format=" LT [EDT]"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-01-05 19:00:00
        </t>,
      ],
      "className": "channel-time",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ago": false,
          "calendar": false,
          "children": "2019-01-05 19:00:00",
          "decimal": false,
          "element": "time",
          "filter": [Function],
          "format": "LT [-]",
          "fromNow": false,
          "interval": 60000,
          "onChange": [Function],
          "titleFormat": "",
          "toNow": false,
          "unit": null,
          "unix": false,
          "utc": false,
          "withTitle": false,
        },
        "ref": null,
        "rendered": "2019-01-05 19:00:00",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "add": Object {
            "hours": 1,
          },
          "ago": false,
          "calendar": false,
          "children": "2019-01-05 19:00:00",
          "decimal": false,
          "element": "time",
          "filter": [Function],
          "format": " LT [EDT]",
          "fromNow": false,
          "interval": 60000,
          "onChange": [Function],
          "titleFormat": "",
          "toNow": false,
          "unit": null,
          "unix": false,
          "utc": false,
          "withTitle": false,
        },
        "ref": null,
        "rendered": "2019-01-05 19:00:00",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="LT [-]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
          <t
            add={
              Object {
                "hours": 1,
              }
            }
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format=" LT [EDT]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
        ],
        "className": "channel-time",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": "LT [-]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "add": Object {
              "hours": 1,
            },
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": " LT [EDT]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="LT [-]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
          <t
            add={
              Object {
                "hours": 1,
              }
            }
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format=" LT [EDT]"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05 19:00:00
          </t>,
        ],
        "className": "channel-time",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": "LT [-]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "add": Object {
              "hours": 1,
            },
            "ago": false,
            "calendar": false,
            "children": "2019-01-05 19:00:00",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": " LT [EDT]",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05 19:00:00",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
}
`;
